packages/react-core/src/components/Icon/Icon.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import styles from '@patternfly/react-styles/css/components/Icon/icon';
+import { css } from '@patternfly/react-styles';
+
+export interface IconComponentProps extends React.HTMLProps<HTMLSpanElement> {
+  /** Icon content */
+  children?: React.ReactNode;
+  /** Progress icon content */
+  progressIcon?: React.ReactNode;
+  /** Additional classes applied to the icon container */
+  className?: string;
+  /** Size of the icon */
+  containerSize?: 'sm' | 'md' | 'lg' | 'xl';
+  /** Size of icon content */
+  contentSize?: 'sm' | 'md' | 'lg' | 'xl';
```

Comment:
I think these prop names work... just some thoughts.

* `containerSize` sets both the outer icon box/container height/width and the icon `font-size`, so that they match. All of these are [`--pf-global--icon--FontSize--md`](https://github.com/patternfly/patternfly/blob/a880484eeb291eb129eed4374f789b6388b11b79/src/patternfly/sass-utilities/scss-variables.scss#L194) by default (same as the "md" size variations in this component)
* `contentSize` sets the icon `font-size`
* `isInline` (below) sets the box/container and font-size to inherit whatever the text size is (also sets the color to inherit from the text color)

It's worth noting that:
* `contentSize` will override the icon size set by `containerSize`
* `containerSize` and `contentSize` will override the sizes if used with `isInline`

I'm wondering if maybe `containerSize` should be `size` and `contentSize` should be `iconSize`? Curious others' thoughts. cc @srambach

Reply:
Yeah, we went back and forth a little on what to call it initially but decided against using "container" so I like `size` for sure for the overall size, and then either `contentSize` or `iconSize` for the inner size seems ok to me.

Reply:
@kmcfaul can you add more detail to the prop descriptions for these size props.  I think it is important to not "contentSize will override the icon size set by containerSize" as stated above.